### PR TITLE
USAGOV-784-defacement-recovery-docs-update

### PR DESCRIPTION
**Pull request by ghost**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-784

## Description

Update docs to a more practical form, hopefully.

Scripts that rely on 'cf ssh' check if it is disabled in the current space before attempting ssh calls.

## Checklist for the Developer

- [ ] A link to the JIRA ticket has been added above.
- [ ] The JIRA ticket identifies the desired result of this change.
- [ ] The JIRA ticket contains clear acceptance criteria.
- [ ] The JIRA ticket contains clear testing steps.
- [ ] The JIRA ticket contains a description of "Done".
- [ ] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [ ] The branch name of this PR matches the project standards.
- [ ] QA steps are followed and any changes to process are updated in the ticket.
- [ ] Code Standards are followed, there are no bad practices in use.
- [ ] Config changes (if any) include only necessary changes.
- [ ] No errors in Drupal or Client Browser.
- [ ] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [ ] There are no known side-effects outside of expected behavior.

### CODE REVIEW

File: bin/snapshot-backups/README-DR.md [hunk 1]
```diff
@@ -1,7 +1,110 @@
...
+
+        export BRANCH=USAGOV-999
+        export SPACE=prod
+        export SUFFIX=pre-deploy
+
+1. Proceed to *Static site backup* step
+
+### Static site backup
+
+        echo=echo   ### Do NOT export echo. Ask me how I know this (tm).
```
Comment (akf) [inline bin/snapshot-backups/README-DR.md:32]: I'm running zsh over here. It turns out that even if I don't export echo, it gets picked up by these scripts. And these instructions confused me a bit -- why not dispense with "echo=echo" and just use "echo" if I want to see the command I'm about to run? 

(Also, I got fixated on "echo" and failed to clear "dryrun" to actually run the script which is my problem.)

File: bin/snapshot-backups/README-DR.md [hunk 2]
```diff
@@ -1,7 +1,110 @@
...
+        dryrun='--dryrun'
+        $echo bin/snapshot-backups/stw ${dryrun} $SPACE $BRANCH $SUFFIX site-snapshot-create
```
Comment (akf) [inline bin/snapshot-backups/README-DR.md:34]: stw isn't working correctly for me. I'll switch shells and see if it helps, but here's what I'm seeing: I get a space before the container tag number.

`
amykfarrell@FCOH2J-T6L7Q05P usagov-2021 % dryrun='--dryrun'
amykfarrell@FCOH2J-T6L7Q05P usagov-2021 % $echo bin/snapshot-backups/stw ${dryrun} $SPACE $BRANCH $SUFFIX site-snapshot-create
bin/snapshot-backups/stw --dryrun dev USAGOV-784 test1 site-snapshot-create
amykfarrell@FCOH2J-T6L7Q05P usagov-2021 % bin/snapshot-backups/stw ${dryrun} $SPACE $BRANCH $SUFFIX site-snapshot-create

Running '/Users/amykfarrell/dev/usagov-2021/bin/snapshot-backups/site-snapshot-create' with tag:  USAGOV-784.dev. 4738.test1

`

Comment (akf) [reply bin/snapshot-backups/README-DR.md:34]: I barreled forward and got a snapshot named "USAGOV-784.dev.". :-)

Comment (akf) [reply bin/snapshot-backups/README-DR.md:34]: Same behavior with bash: 

`amykfarrell@FCOH2J-T6L7Q05P ~ % /bin/sh --version
GNU bash, version 3.2.57(1)-release (arm64-apple-darwin21)
Copyright (C) 2007 Free Software Foundation, Inc.
amykfarrell@FCOH2J-T6L7Q05P ~ % /bin/sh
sh-3.2$ cd dev/usagov-2021/
sh-3.2$ dryrun='--dryrun'
sh-3.2$ export BRANCH=USAGOV-784
sh-3.2$ export SPACE=dev
sh-3.2$ export SUFFIX=try2
sh-3.2$ bin/snapshot-backups/stw ${dryrun} $SPACE $BRANCH $SUFFIX site-snapshot-create

Running '/Users/amykfarrell/dev/usagov-2021/bin/snapshot-backups/site-snapshot-create' with tag:  USAGOV-784.dev. 4738.try2

/Users/amykfarrell/dev/usagov-2021/bin/snapshot-backups/site-snapshot-create USAGOV-784.dev. 4738.try2
`

Comment (akf) [reply bin/snapshot-backups/README-DR.md:34]: The culprit here is spaceCCIContainerTag in bin/deploy/includes. I'm looking into why it's failing to trim the leading space from the container tag.

Comment (akf) [reply bin/snapshot-backups/README-DR.md:34]: Better now, thanks!

File: bin/snapshot-backups/README-DR.md [hunk 3]
```diff
@@ -1,7 +1,110 @@
...
+
+### Example
+
+        bin/snapshot-backups/stw prod USAGOV-787 pre-deploy site-snapshot-create
+
+See the file
+
+        bin/deploy/includes
+
+Specifically the functions *assertSpace,  spaceCCIContainerTag* and *createSpaceAssertedBackupTag* for details of how the *stw* (snapshoot tool wrapper) script assembles the backup tag, and asserts that the currect space matches the arguments provided to *stw*
```
Comment (akf) [inline bin/snapshot-backups/README-DR.md]: snapshoot -> snapshot

File: bin/snapshot-backups/README-DR.md [hunk 4]
```diff
@@ -12,33 +115,22 @@
         For example:
         USAGOV-784-defacement-recovery.dev.4250.process_test_001
 
-### Static site backup
-        export echo=echo
+### Manually Tagged Static site backup
+
+        echo=echo   ### Do NOT export echo. Ask me how I know this (tm).
+        dryrun='--dryrun'
```
Comment (akf) [inline bin/snapshot-backups/README-DR.md:121]: Just to be picky, let's either remove "dryrun='--dryrun'" from these last couple of examples, or actually use ${dryrun} in the examples.

File: bin/snapshot-backups/README-DR.md [hunk 5]
```diff
@@ -12,33 +115,22 @@
...
-
-### DB backup
-        export echo=echo
-        $echo bin/snapshot-backups/stw $BACKUP_TAG db-dump-download
-        $echo bin/snapshot-backups/stw $BACKUP_TAG db-dump-push-to-snapshot
-        $echo bin/snapshot-backups/db-list
 ## 3. Snapshot restore
 
-## 4. Snapshot restore using helper
+TBD
```
Comment (akf) [inline bin/snapshot-backups/README-DR.md]: Is this actually not implemented (site-snapshot-deploy looks promising), or just not documented?

**Review by akf — CHANGES_REQUESTED** [review]
I added a bunch of comments, but the important ones are: 
* is snapshot restore not implemented, or just not documented, and could that be included in this PR? 
* Let's talk about my confusion over what you're suggesting with echo.

File: bin/snapshot-backups/README-DR.md (hunk 3 above)
Comment (arpage) [reply bin/snapshot-backups/README-DR.md]: lol - fixed

File: bin/snapshot-backups/README-DR.md (hunk 5 above)
Comment (arpage) [reply bin/snapshot-backups/README-DR.md]: Just not documented yet.

File: bin/snapshot-backups/README-DR.md (hunk 4 above)
Comment (arpage) [reply bin/snapshot-backups/README-DR.md:121]: good call - added.

**Review by akf — APPROVED** [review]

**Pull request merged**